src/gpu/sycl/sycl_primitive_conf.hpp

```diff
@@ -52,9 +52,11 @@ struct sycl_binary_conf_t {
 struct sycl_eltwise_conf_t {
     prop_kind_t prop_kind;
     xpu::sycl::md_t src_md;
-    xpu::sycl::md_t dst_md;
-    xpu::sycl::md_t diff_src_md;
-    xpu::sycl::md_t diff_dst_md;
+    data_type_t dst_dt;
+    xpu::sycl::md_t::dims32_t dst_dims;
+    xpu::sycl::md_t::dim32_t dst_ndims;
+    data_type_t diff_src_dt;
```

Comment:
Just in case, what would happen if formats of `diff_dst` and `diff_src` and `src`/`dst` are different? Wouldn't they all be needed?

Reply:
Based on the `src_d == dst_d` check here https://github.com/oneapi-src/oneDNN/blob/main/src/gpu/sycl/ref_eltwise.hpp#L50 I think it is not supported currently. Additionally, in the kernel itself dst is indexed with the same index calculated for src. 

I am assuming this wasn't caught before because the benchdnn eltwise driver accepts `--tag` which applies the same format to both src and dst.

Reply:
While the statement is true as of now, it may change, since the API allows different md for src/dst/diff_src/diff_dst. The question is why to remove the functionality now when it might be needed to be restored?

Reply:
@dzarukin I looked into this again and noticed that the binary post op input array is larger than the supported number of max post ops and the kernel itself only accesses 5 elements from the array. I reduced the size of the array and that made the conf struct small enough to workaround the argument size limit issue. So I reverted the previous changes.